lxd/images.go

```diff
@@ -304,21 +304,95 @@ var imagesCmd = Command{name: "images", post: imagesPost, get: imagesGet}
 func imageDelete(d *Daemon, r *http.Request) Response {
 	shared.Debugf("responding to image:delete")
 
-	uuid := mux.Vars(r)["name"]
-	uuidfname := shared.VarPath("images", uuid)
-	err := os.Remove(uuidfname)
+	fingerprint := mux.Vars(r)["fingerprint"]
+	fname := shared.VarPath("images", fingerprint)
+	err := os.Remove(fname)
 	if err != nil {
-		shared.Debugf("Error deleting image file %s: %s\n", uuidfname, err)
+		shared.Debugf("Error deleting image file %s: %s\n", fname, err)
 	}
 
-	_, _ = d.db.Exec("DELETE FROM images_aliases WHERE image_id=(SELECT id FROM images WHERE fingerprint=?);", uuid)
-	_, _ = d.db.Exec("DELETE FROM images_properties WHERE image_id=(SELECT id FROM images WHERE fingerprint=?);", uuid)
-	_, _ = d.db.Exec("DELETE FROM images WHERE fingerprint=?", uuid)
+	_, _ = d.db.Exec("DELETE FROM images_aliases WHERE image_id=(SELECT id FROM images WHERE fingerprint=?);", fingerprint)
+	_, _ = d.db.Exec("DELETE FROM images_properties WHERE image_id=(SELECT id FROM images WHERE fingerprint=?);", fingerprint)
+	_, _ = d.db.Exec("DELETE FROM images WHERE fingerprint=?", fingerprint)
```

Comment:
I know it's not getting us much, but would you mind doing the "SELECT id from IMAGES where fingerprint=?" separately and then use the value in the two DELETEs? That'd save us one SQL query.